packages/react/src/components/DateTimePicker/DateTimePicker.jsx

```diff
@@ -477,8 +498,96 @@ const DateTimePicker = ({
     [absoluteValue, relativeValue]
   );
 
-  const onFieldClick = () => {
-    setIsExpanded(!isExpanded);
+  const getFocusableSiblings = () => {
+    if (presetListRef?.current) {
+      const siblings = presetListRef.current.querySelectorAll('[tabindex]');
+      return Array.from(siblings).filter(
+        (sibling) => parseInt(sibling.getAttribute('tabindex'), 10) !== -1
+      );
+    }
+
+    return [];
+  };
+
+  const onFieldInteraction = ({ key }) => {
+    switch (key) {
+      case 'Escape':
+        setIsExpanded(false);
+        break;
+      // if the input box is focused and a down arrow is pressed this
+      // moves focus to the first item in the preset list that has a tabindex
+      case 'ArrowDown':
+        if (presetListRef?.current) {
+          const listItems = getFocusableSiblings();
+          if (listItems?.[0]?.focus) {
+            listItems[0].focus();
+          }
+        }
+        break;
+      default:
+        setIsExpanded(!isExpanded);
+        break;
+    }
+  };
+
+  /**
+   * Moves up the preset list to the previous focusable element or wraps around to the bottom
+   * if already at the top.
+   */
+  const moveToPreviousElement = () => {
```

Comment:
Could these be helper utilities that we can pull out so they can be used elsewhere? Or are these specific to this component?

Reply:
theoretically, I suppose they could. I don't know exactly where we would use it, but it's only looking for siblings with `tabindex != -1` and wrapping around the ends of the array to move through them...